exonum/src/storage/proof_map_index/key.rs

```diff
@@ -130,142 +189,67 @@ impl DBKey {
         }
     }
 
-    /// Shortens this DBKey to the specified length.
-    pub fn prefix(&self, suffix: u16) -> DBKey {
-        debug_assert!(self.from + suffix <= self.data.len() as u16 * 8);
+    fn start_from(&self, start: u16) -> Self {
+        debug_assert!(start <= self.end());
 
-        DBKey {
-            data: self.data,
-            from: self.from,
-            to: self.from + suffix,
-        }
+        let mut key = ProofPath::from_raw(self.bytes);
+        key.start = start;
+        key
     }
 
-    /// Return object which represents a view on to this slice (further) offset by `i` bits.
-    pub fn suffix(&self, suffix: u16) -> DBKey {
-        debug_assert!(self.from + suffix <= self.to);
+    fn prefix(&self, pos: u16) -> Self {
+        debug_assert!(self.start() + pos <= self.raw_key().len() as u16 * 8);
 
-        DBKey {
-            data: self.data,
-            from: self.from + suffix,
-            to: self.to,
-        }
+        let mut key = ProofPath::from_raw(self.bytes);
+        key.start = self.start;
+        key.set_end(self.start + pos);
+        key
     }
 
-    /// Shortens this DBKey to the specified length.
-    pub fn truncate(&self, size: u16) -> DBKey {
-        debug_assert!(self.from + size <= self.to);
-
-        DBKey {
-            data: self.data,
-            from: self.from,
-            to: self.from + size,
-        }
+    fn suffix(&self, pos: u16) -> Self {
+        self.start_from(self.start() + pos)
     }
 
-    /// Returns how many bits at the beginning matches with `other`
-    pub fn common_prefix(&self, other: &Self) -> u16 {
+    fn common_prefix(&self, other: &Self) -> u16 {
         // We assume that all slices created from byte arrays with the same length
```

Comment:
Nitpick: this comment doesn't make much sense to me, and if we specify that `common_prefix` should work only with ranges starting from the same position, it could be safely removed.